JoinColony/colonyCDapp
# Feat: Handle the cancelAndWaive method call for Streaming Payments

## Description

Handle the `cancelAndWaive` method call for Streaming Payments.

Here are the required block-ingestor updates: [block-ingestor#239: handle the StreamingPayments contract's ClaimWaived event emission](https://github.com/JoinColony/block-ingestor/pull/239).

![2217](https://github.com/JoinColony/colonyCDapp/assets/50642296/5ec52530-d095-4928-8d89-4dd6771e890a)

## Testing

1. Enable the Streaming Payments extension
2. Go to your Colony's dashboard and using the debug UI, create a Streaming Payment
![create_streaming_payment](https://github.com/JoinColony/colonyCDapp/assets/50642296/e5f6f399-9f26-4cdd-9c27-2b8ae020da78)
3. Open the GraphQL Playground http://localhost:20002 and run the following query:
> This query will show all streaming payments
```
query MyQuery {
  listStreamingPayments {
    items {
      isWaived
      isCancelled
      amount
      nativeId
    }
  }
}
``` 
4. Verify that your streaming payment's `isCancelled` and `isWaived` fields are both set to `null`:
```
{
  "isWaived": null,
  "isCancelled": null,
  "amount": "1000000000000000000",
  "nativeId": 5 // your nativeId will vary
},
```
5. Press the "Cancel and Waive" button
![cancel_and_waive_button](https://github.com/JoinColony/colonyCDapp/assets/50642296/37e7ef25-7cc9-4c1f-9a30-bfb3e8ecdd4b)
6. Run the query in Step 3 and verify that your streaming payment's `isCancelled` and `isWaived` fields are both set to `true`
```
{
  "isWaived": true,
  "isCancelled": true,
  "amount": "1000000000000000000",
  "nativeId": 5
},
```

## Diffs

**New stuff** ✨

* ~A new saga called `cancelAndWaiveStreamingPaymentAction` to handle the `cancelAndWaive` action for a streaming payment. I could have repurposed the existing `cancelStreamingPayment` saga but then I'd also have to litter it with a lot of `if that, else that` statements because we're not accepting annotations for the `cancelAndWaive` actio, at least for now, which means I won't need a Colony Manager Client and an extra Channel for the annotation upload. It also makes it easier to read.~

P.S. ~I could already sense @rdig asking and yes, this one will likely need a migration 👀~ Jakub just explained to me that streaming payments is not actually a live feature yet so no migration required, IGNORE ME!

Resolves #2217
